applications/CompressiblePotentialFlowApplication/tests/cpp_tests/test_transonic_perturbation_potential_element.cpp

```diff
@@ -426,6 +437,99 @@ KRATOS_TEST_CASE_IN_SUITE(PingTransonicPerturbationPotentialFlowElementLHS, Comp
     }
 }
 
+/** Checks the TransonicPerturbationPotentialFlowElement.
+ * Tests the LHS computation.
+ */
+KRATOS_TEST_CASE_IN_SUITE(PingTransonicPerturbationPotentialFlowSupersonicElementLHS, CompressiblePotentialApplicationFastSuite) {
+    Model this_model;
+    ModelPart& model_part = this_model.CreateModelPart("Main", 3);
+
+    GenerateTransonicPerturbationElement(model_part);
+    GenerateTransonicPerturbationUpwindElement(model_part);
+
+    Element::Pointer pElement = model_part.pGetElement(1);
+    Element::Pointer pUpwindElement = model_part.pGetElement(2);
+    const unsigned int number_of_nodes = pElement->GetGeometry().size();
+
+    FindNodalNeighboursProcess find_nodal_neighbours_process(model_part);
+    find_nodal_neighbours_process.Execute();
+
+    pElement->Initialize(model_part.GetProcessInfo());
+    pUpwindElement->SetFlags(INLET);
+
+    // velocity[0] : 394.18
+    // velocity[1] : -50
+    // local_mach_number_squared : 1.705
+    AssignPotentialsToSupersonicTransonicPerturbationElement(pElement);
+
+    // upwind_velocity[0] : 154.18
+    // upwind_velocity[1] : 149
+    // upwind_mach_number_squared : 0.3999
+    AssignPotentialsToNormalTransonicPerturbationElement(pUpwindElement);
```

Comment:
In AssignPotentialsToSupersonicTransonicPerturbationElement you assign the values 10, 200 and 150 and to nodes 1, 2 and 3 respectively and in AssignPotentialsToNormalTransonicPerturbationElement you assign the values 1, 100 and 150 to nodes 1, 3 and 4 respectively, overwriting the previous values on the shared nodes 1 and 3. It might be easier to follow to define the potential values when calling the function, for example:

```cpp
    std::array<double, 3> potential{10.0, 200.0, 150.0};
    AssignPerturbationPotentialsToTransonicElement(*pElement, potential);
```

where AssignPerturbationPotentialsToTransonicElement is:

```cpp
void AssignPerturbationPotentialsToTransonicElement(Element& rElement, const std::array<double, 3> rPotential) {
    for (unsigned int i = 0; i < 3; i++){
        rElement.GetGeometry()[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL) = rPotential[i];
    }
}
```